storage/scratchpad/src/sparse_merkle/updater.rs

```diff
@@ -265,12 +265,24 @@ impl<'a, V: Send + Sync + Clone + CryptoHash> SubTreeUpdater<'a, V> {
     }
 
     fn run(self, proof_reader: &impl ProofRead<V>) -> Result<InMemSubTreeInfo<V>> {
+        // Limit total tasks that are potentially sent to other threads.
+        const MAX_PARALLELIZABLE_DEPTH: usize = 8;
```

Comment:
could/should we make this a function of rayon::current_num_threads()?

Reply:
Probably a few times of that, because the work pieces can differ in size. -- Definitely something we can keep digging and tuning, but I opt for a straightforward way for now. 

I spent a few day to balance out things and sent out this short diff, but can still see it performs extremely unsatisfying in cases I didn't test, for example when other components got parallelized and race for threads. -- Again, I think pipelining from higher level deserves more love than fine tuning here.